GeneralsMD/Code/GameEngine/Source/GameLogic/Object/Update/DockUpdate/SupplyCenterDockUpdate.cpp

```diff
@@ -129,14 +130,8 @@ Bool SupplyCenterDockUpdate::action( Object* docker, Object *drone )
 	}
 
 	Bool displayMoney = value > 0 ? TRUE : FALSE;
-	if( getObject()->testStatus(OBJECT_STATUS_STEALTHED) )
-	{
-		// OY LOOK!  I AM USING LOCAL PLAYER.  Do not put anything other than TheInGameUI->addFloatingText in the block this controls!!!
-		if( !getObject()->isLocallyControlled() && !getObject()->testStatus(OBJECT_STATUS_DETECTED) )
-		{
-			displayMoney = FALSE;
-		}
-	}
+	if (!getObject()->getDrawable()->isVisible())
+		displayMoney = FALSE;
 
 	if( displayMoney )
```

Comment:
Can simplify to `value > 0 && getObject()->getDrawable()->isVisible()`.

Twice.

Reply:
Hmm, alright. I thought this way was slightly more readable and resulted in a cleaner diff.